mongodb/resource_automation_agent.go

```diff
@@ -47,6 +47,9 @@ func resourceMdbAutomationAgentCreate(data *schema.ResourceData, meta interface{
 		return fmt.Errorf("could not create a SSH client: %v", err)
 	}
 
+	// install curl
+	ssh.PanicOnError(client.RunCommand(conn.SudoPrefix("apt-get install curl")))
```

Comment:
Sorry, I wasn't explicit enough!  Installing `curl` (and any other system libraries the provider needs) should be added to the `Dockerfile`.
We can set a hard requirement that any target hosts should have a few libraries available for this terraform-provider to work. And we can use that list to write documentation for this codebase (at some point.)

There are a few potential issues with running apt, which increase the possibility of running into issues:
- apt jobs can run at any time: if they run at the same time as the provider, they will cause this call to fail
- we use `SudoPrefix` currently, but we shouldn't (the `chef` terraform provisioner allows the user to specify if use_sudo is allowed - that's a much more secure approach as having root rights on a host means you can do anything you want, but we shouldn't really need sudo at all)
- in a future version, we'll support installing Ops Manager from tar.gz, thus completely removing the need for `root` permissions

Reply:
That makes sense! I think curl is a reasonable library to require too, plus the instructions from OM for doing it semi-automated also require curl.